app/views/questionnaire.py

```diff
@@ -304,7 +304,10 @@ def _render_template(current_location, context=None, template=None):
     previous_location = path_finder.get_previous_location(current_location)
 
     previous_url = None
-    if previous_location is not None and not SchemaHelper.is_first_block_id_for_group(g.schema_json, current_location.group_id, current_location.block_id):
+
+    is_first_block_for_group = SchemaHelper.is_first_block_id_for_group(g.schema_json, current_location.group_id, current_location.block_id)
+
+    if previous_location is not None and not is_first_block_for_group and not current_location.block_id == 'thank-you':
```

Comment:
A more general observation, I wonder if we should move the previous location logic into the navigation class after the previous location is a part of navigation even if it's not rendered in the same way